test/cases/compile_errors/compare_optional_to_non-optional_with_invalid_types.zig

```diff
@@ -1,4 +1,4 @@
-export fn inconsistentChildType() void {
+export fn compatibleChildType() void {
```

Comment:
This test case is intended to test for compile errors. Please remove this function from this case, and add a corresponding behavior test case instead for is combination of types.

Reply:
Okay - there are other tests in this file which intentionally don't error, so was following that example. Should I transition those to behaviour instead?